scripts/staticcheck.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+echo "==> Checking that code complies with static analysis requirements..."
+# The legacy code is frozen, and should not be updated. It will be removed once
+# we can refactor the remote backends to no longer require it.
+skip="internal/legacy|backend/remote-state/|internal/planproto|internal/tfplugin"
```

Comment:
The above comment reads as if `internal/planproto` and `internal/tfplugin` fit in the category of "legacy code that will be removed", but I assume that isn't what you intended. To minimize confusion for future contributors, can we say something about excluding generated code in that comment too?

Reply:
thanks, that was updated from the old commit, but the comment is out of date. I'll re-word

Reply:
It might also be clearer to write out `tfplugin5` and `tfplugin6` separately here so that we can see exactly which packages are being excluded. :thinking: